Lib/ctypes/test/test_cast.py

```diff
@@ -82,5 +82,18 @@ def test_wchar_p(self):
         self.assertEqual(cast(cast(s, c_void_p), c_wchar_p).value,
                              "hiho")
 
+    def test_bad_type_arg(self):
+        # The type argument must be a ctypes pointer type.
+        array_type = c_byte * sizeof(c_int)
+        array = array_type()
+        self.assertRaises(TypeError, cast, array, None)
+        self.assertRaises(TypeError, cast, array, array_type)
+        class Struct(Structure):
```

Comment:
I don't know if cpython does this, but linking to the bug number in the test might be nice.